tests/test_api_profile.py

```
@@ -0,0 +1,407 @@
+        env  = make_env(db=MockDB([make_stmt(first=row), acts]))
+        r    = await worker.api_get_public_profile("alice", self._req(), env)
+        d    = _parse(r)["data"]
+        assert d["is_teacher"] == 1
+
+
+# ---------------------------------------------------------------------------
+# POST /api/profile/avatar  (avatar upload validation)
+# ---------------------------------------------------------------------------
+
+class TestUploadAvatar:
+    def _req(self, payload, headers=None):
```

Ruff (0.15.15)

[warning] Missing return type annotation for private function `_req`

(ANN202)